airflow/models/base.py

```diff
@@ -44,6 +44,19 @@ def get_id_collation_args():
     if collation:
         return {'collation': collation}
     else:
+        # Automatically use utf8mb3_general_ci collation for mysql
+        # This is backwards-compatible. All our IDS are ASCII anyway so even if
+        # we migrate from previously installed database with different collation and we end up mixture of
+        # COLLATIONS, it's not a problem whatsoever (and we keep it small enough so that our indexes
+        # for MYSQL will not exceed the maximum index size.
+        #
+        # See https://github.com/apache/airflow/pull/17603#issuecomment-901121618.
+        #
+        # We cannot use session/dialect as at this point we are trying to determine the right connection
+        # parameters, so we use the connection
+        conn = conf.get('core', 'sql_alchemy_conn', fallback='')
+        if conn.startswith('mysql'):
```

Comment:
```suggestion
        if conn.startswith('mysql') or conn.startswith('mariadb'):
```

Not that it's officially supported, but with #16907 open, doesn't hurt?

Reply:
Would `conn.dialect.name == "mysql"` be more robust?

Reply:
Conn. is just a string. We do not want to instantiate a session just yet at this moment, wy just want to use connection string to make a decision here!

Reply:
> Not that it's officially supported, but with #16907 open, doesn't hurt?

Agree :)